functions/go/set-namespace/namespace_transformer.go

```diff
@@ -60,6 +71,67 @@ func (p *plugin) Transform(m resmap.ResMap) error {
 		if err != nil {
 			return err
 		}
+		if err = p.roleBindingHack(r, sAL); err != nil {
+			return err
+		}
 	}
 	return nil
 }
+
+// roleBindingHack extends the behavior of the upstream kustomize filter
+// This extension adds the following behavior:
+// Given a ServiceAccount is present in the ResourceList, additionally set the
+// namespace for that ServiceAccount where it is referenced in the "subjects"
+// field of a RoleBinding or ClusterRoleBinding
+func (p *plugin) roleBindingHack(r *resource.Resource, sAL ServiceAccountLookup) error {
+	kind := r.GetKind()
+	if kind != roleBindingKind && kind != clusterRoleBindingKind {
+		return nil
+	}
+	subjects, err := r.Pipe(kyaml.Lookup(subjectsField))
+	if err != nil || kyaml.IsMissingOrNull(subjects) {
+		return err
+	}
+
+	err = subjects.VisitElements(func(o *kyaml.RNode) error {
+		subjectKind := o.GetKind()
+		if subjectKind != serviceAccountKind {
+			return nil
+		}
+		var nameRN *kyaml.RNode
+		nameRN, err = o.Pipe(kyaml.Lookup("name"))
+		if err != nil || kyaml.IsMissingOrNull(nameRN) {
+			return err
```

Comment:
This case is handled the same as the `subjects` field above, where if there is no error but the field is missing/null then it will return `nil`. I did not think it would be preferred to swallow the error here, but WDYT?